### ISSUE TITLE
Chat priorities: EssentialsX ignore, AdancedBan mute...

### ISSUE THREAD

**Issue by MithrandirCraft** [kangarko/ChatControl-Pro#1427]
```
Listener_Priority:
  Formatter: HIGH
  Checker: NORMAL
  Signs: HIGHEST
```

- EssentialsX: You can read ignored (with essentialsX) player's messages in chat.
- AdvancedBan: Muted players can still speak. (This one was fixed, probably by it's developer by increasing it's own priority maybe...) I'm only commenting on this one (Which I already reported once in the past) because it's pretty much the same, and as another example. 

Setting the Formatter priority to NORMAL doesn't fix EssentialsX ignoring. And it causes double messages if using chat channels + /staffchat, as you can see here:
![imagen](https://user-images.githubusercontent.com/6623704/67131362-b7442c00-f204-11e9-8212-19221662d233.png)

If the Formatter priority is set to LOW or LOWEST, and the plugin is loaded/reloaded, these errors show up: https://pastebin.com/Led2gWGX
I know it says do not report, but you're telling me in this error (from what I can gather from the .log since it's not very readable through notepad++) is that I should leave formatting priority to HIGHEST. But I think I need to set it lower, otherwise these issues emerge.

There are no more priority levels, so I can't fix the essentials ignore issue.

I would really appreciate some indications on how to fix this apparent ignore priority issue. Maybe I should just ask EssentialsX to increase their own priority.

**Comment by kangarko** [comment]
Hi,

thank you for reporting. Previously we enabled setting Formatter priority to LOWEST (ie lower than Checker) but that opened numerous other problems so it is now disabled.

I am afraid I cannot do anything more. You can choose to use ChatControl's ignore system (that will read EssentialsX user data to be compatible) instead. Or you can ask their dev team to increase the priority or you can set Checker to LOW and Formatter to NORMAL to test if any results will vary.

**Comment by kangarko** [comment]
Closing due to inactivity, return back below if you need more help from us

**Comment by MithrandirCraft** [comment]
I mean, if it solves it, I wouldn't mind to use CC's ignore. If it's an actual simplified command that players can use for ignoring players.

Is there any information on such a feature?
Also the answer I got from mentioning this issue to essentials is quite extense: https://github.com/EssentialsX/Essentials/issues/2837

**Comment by kangarko** [comment]
Thank you. We are listening to that event on whatever priority you configure in the settings.yml, but from what they have said, they will always be first so it should work. We even do check for EssentialsX ignore system directly in ChatControl.

When I reviewed the code I found that unless Ignore.Enabled is on true in settings.yml, the entire ignore system in ChatControl is shut down, thus no checks occur.

And since we send messages in an interactive way (such as the hover and click elements) that Bukkit does not support, we have to clear all recipients and send the message in an own way.

Having said that, if you want to use the ignore system from either plugin, enable Ignore.Enabled in settings.yml. This will make ChatControl take over EssentialsX, but existing ignores will be preserved.
